tests/Test/AbstractFixerTestCase.php

```diff
@@ -461,7 +461,9 @@ private function getLinter(): LinterInterface
 
         if (null === $linter) {
             $linter = new CachingLinter(
-                getenv('FAST_LINT_TEST_CASES') ? new Linter() : new ProcessLinter()
+                filter_var(getenv('FAST_LINT_TEST_CASES'), FILTER_VALIDATE_BOOLEAN)
```

Comment:
BC breaker. imagine someone was doing `FAST_LINT_TEST_CASES=x`

Reply:
In my opinion it's rather bugfix, because previous behavior was "truish" for any non-empty and non-0 value. I've added it because fallbacking in CI definition using `||` is triggered even if value is explicitly set, but has 0 value (that's why ,`no` is used). I wouldn't consider it as BC-break.

Reply:
if to consider it a bugfix, then we should call a bug to be a missing validation. but now we don't really validate properly, if input is not bolleanish, we just ignore it without telling that to the user.
Then it's even easier to have the validation error msg / deprecation message combined

Reply:
> if input is not bolleanish, we just ignore it without telling that to the user

Should we really focus that much on [hidden feature](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/5206)? 🤷 

In worst scenario tests will take longer on one's side if one's using `FAST_LINT_TEST_CASES` with non-booleanish value, that currently works, and it's easy to fix. [Github search](https://github.com/search?q=FAST_LINT_TEST_CASES&type=code) shows that this flag is barely used - mostly in our internal code and forks, and when it's actually used, it has `0`/`1` values.

Reply:
I imagine if it's used, it's on local dev envs.

i agree we should first solve this comment - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6890#pullrequestreview-1376795096 - but for validation, we shouldn't silently ignore the errors ;)